variable/comparison.cpp

```diff
@@ -14,12 +15,39 @@ using namespace scipp::core;
 namespace scipp::variable {
 
 namespace {
-Variable _values(Variable &&in) { return in.hasVariances() ? values(in) : in; }
+Variable _values(const Variable &&in) {
+  return in.hasVariances() ? values(in) : in;
+}
+
+Variable tolerance(const Variable &atol, const Variable &rtol,
+                   const Variable &y) {
+  return atol + rtol * abs(y);
+}
 } // namespace
 
+Variable allclose(const Variable &a, const Variable &b, const Variable &rtol,
+                  const Variable &atol, const NanComparisons equal_nans) {
```

Comment:
I describe that option in the PR description - That was my initial thought, but would be slower and use more memory. I think `allclose` will primarily be used in testing, but It's also possible that users will want it for branching or stop condition for iteration etc, which is why I implemented as a reduction operation.

Reply:
- Premature optimization
- More risk for bugs
- Potentially worse for multi-threading, since multi-dim `accumulate_in_place` into single element might not be threaded as well as stepwise, e.g., 2d->1d->0d as performed in `all`.
- More relevant optimization (if it would matter) would be to avoid the *two* copies by `isclose` to `values` and `variances`, instead of the single copy avoided by this, but even that is not worth doing now, I think.

Reply:
See last item here: https://github.com/scipp/scipp/blob/e873502d5ce3ca6e482bb8a2d93ffc56b913c3af/variable/include/scipp/variable/accumulate.h#L17-L21

That is, the manual "optimization" may actually be a pessimization in this case.

Reply:
On my macbook. For 1d inputs ranging 10^0 <= size < 10^6 in length
![image](https://user-images.githubusercontent.com/1109536/122107585-22e3ff80-ce13-11eb-919b-10cbe29d785c.png)
On my macbook. For 2d inputs ranging 10^0 <= size < 10^6 in length (in x, y)
![image](https://user-images.githubusercontent.com/1109536/122107878-7c4c2e80-ce13-11eb-8ad7-42a3d46a6cc0.png)